### PR TITLE
Add table aws_glue_job. Closes #1096

### LINKED ISSUE #1096: Add table  aws_glue_job
**References**
https://docs.aws.amazon.com/sdk-for-go/api/service/glue/#Glue.GetJobs
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/glue_job

### PR DESCRIPTION
**Pull request by karanpopat**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Outputs:

aws_account = "123456789012"
aws_partition = "aws"
aws_region = "us-east-1"
resource_aka = "arn:aws:glue:us-east-1:123456789012:job/turbottest59867"
resource_name = "turbottest59867"

Running SQL query: test-get-query.sql
[
  {
    "arn": "arn:aws:glue:us-east-1:123456789012:job/turbottest59867",
    "description": null,
    "name": "turbottest59867",
    "region": "us-east-1"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "arn": "arn:aws:glue:us-east-1:123456789012:job/turbottest59867",
    "description": null,
    "name": "turbottest59867",
    "region": "us-east-1"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:glue:us-east-1:123456789012:job/turbottest59867"
    ],
    "name": "turbottest59867",
    "region": "us-east-1",
    "title": "turbottest59867"
  }
]
✔ PASSED

POSTTEST: tests/aws_glue_job

TEARDOWN: tests/aws_glue_job

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```sql
> select
  name,
  created_on,
  description,
  max_capacity,
  number_of_workers,
  region,
  timeout
from
  aws_glue_job;

+------+---------------------------+-------------+--------------+-------------------+-----------+---------+
| name | created_on                | description | max_capacity | number_of_workers | region    | timeout |
+------+---------------------------+-------------+--------------+-------------------+-----------+---------+
| job1 | 2022-06-28T13:56:39+05:30 |             | 2            | 2                 | us-west-1 | 2880    |
+------+---------------------------+-------------+--------------+-------------------+-----------+---------+

> select
  title,
  arn,
  created_on,
  connections -> 'Connections' as connections
from
  aws_glue_job
where
  connections is not null;
+-------+----------------------------------------------+---------------------------+-------------+
| title | arn                                          | created_on                | connections |
+-------+----------------------------------------------+---------------------------+-------------+
| job1  | arn:aws:glue:us-west-1:123456789012:job/job1 | 2022-06-28T13:56:39+05:30 | ["testcon"] |
+-------+----------------------------------------------+---------------------------+-------------+

> select
  title,
  arn,
  created_on,
  job_bookmark ->> 'Attempt' as total_attempts,
  job_bookmark ->> 'Run' as total_runs,
  job_bookmark ->> 'RunId' as run_id
from
  aws_glue_job
where
   job_bookmark is not null;
+-------+----------------------------------------------+---------------------------+----------------+------------+-------------------------------------------------------------------------------+
| title | arn                                          | created_on                | total_attempts | total_runs | run_id                                                                        |
+-------+----------------------------------------------+---------------------------+----------------+------------+-------------------------------------------------------------------------------+
| job1  | arn:aws:glue:us-west-1:123456789012:job/job1 | 2022-06-28T13:56:39+05:30 | 4              | 1          | jr_bebee0d1655927791bef07a70359d794cf52b8e0db95b3d4eb3acaa2d0342d07_attempt_3 |
+-------+----------------------------------------------+---------------------------+----------------+------------+-------------------------------------------------------------------------------+
```
</details>

### CODE REVIEW

**Review by rajlearner17 — APPROVED** [review]
LGTM

File: aws/table_aws_glue_job.go [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+		Description: "AWS Glue Job",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.SingleColumn("name"),
+			IgnoreConfig: &plugin.IgnoreConfig{
+				ShouldIgnoreErrorFunc: isNotFoundError([]string{"EntityNotFoundException"}),
+			},
+			Hydrate: getGlueJob,
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listGlueJobs,
```
Comment (ParthaI) [inline aws/table_aws_glue_job.go:28]: Do we have any optional quals support for it?

File: docs/tables/aws_glue_job.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+  created_on
+  region,
+  account_id,
+  default_arguments ->> '--encryption-type' as encryption_type
+from
+  aws_glue_job
+where
+  default_arguments ->> '--encryption-type' is null;
+```
+
```
Comment (ParthaI) [inline docs/tables/aws_glue_job.md:163]:
```suggestion
```

File: aws/table_aws_glue_job.go (hunk 1 above)
Comment (karanpopat) [reply aws/table_aws_glue_job.go:28]: Only pagination params are there

**Review by ParthaI — APPROVED** [review]
LGTM

**Pull request merged**